### PR TITLE
TSDocs lint

### PR DESCRIPTION
**Pull request by HarelM**
Fixes all the lint errors.
Added hyphen.
Changed all the `@tags` to be relevant.

### CODE REVIEW

File: src/style/style.ts [hunk 1]
```diff
@@ -1632,10 +1632,10 @@ export class Style extends Evented {
     /**
      * Add a sprite.
      *
-     * @param id id of the desired sprite
-     * @param url url to load the desired sprite from
-     * @param options style setter options
-     * @param completion the completion handler
+     * @param id - The id of the desired sprite
+     * @param url - The url to load the desired sprite from
+     * @param options - Thr style setter options
```
Comment (birkskyum) [inline src/style/style.ts:1637]: typo

File: src/ui/events.ts [hunk 2]
```diff
@@ -273,17 +267,14 @@ export class MapWheelEvent extends Event {
...
     get defaultPrevented(): boolean {
         return this._defaultPrevented;
     }
 
     _defaultPrevented: boolean;
 
-    /**
-     * @private
-     */
+    /** */
```
Comment (birkskyum) [inline src/ui/events.ts:277]: how come this constructor is kept in docs, when all the other are removed?

File: src/ui/events.ts [hunk 3]
```diff
@@ -447,952 +440,869 @@ export type MapEventType = {
...
+ *   console.log('A idle event occurred.');
+ * });
+ * ```
+ *
+ * @event `remove` Fired immediately after the map has been removed with {@link Map#remove}.
+ *
+ * @example
+ * ```ts
+ * // Initialize the map
+ * letmap = new maplibregl.Map({ // map options });
```
Comment (birkskyum) [inline src/ui/events.ts:1131]: space missing `let map`

File: src/ui/events.ts [hunk 4]
```diff
@@ -447,952 +440,869 @@ export type MapEventType = {
...
+ * @event `error` Fired when an error occurs. This is GL JS's primary error reporting
+ * mechanism. We use an event instead of `throw` to better accommodate
+ * asynchronous operations. If no listeners are bound to the `error` event, the
+ * error will be printed to the console.
+ * `data` - `{error: {message: string}}`
+ *
+ * @example
+ * ```ts
+ * // Initialize the map
+ * letmap = new maplibregl.Map({ // map options });
```
Comment (birkskyum) [inline src/ui/events.ts:1148]: space missing

**Review by birkskyum — APPROVED** [review]

File: src/ui/events.ts (hunk 2 above)
Comment (HarelM) [reply src/ui/events.ts:277]: Depends on the links between the classes originating from the index.ts (entry) file I believe.

File: src/ui/events.ts (hunk 4 above)
Comment (HarelM) [reply src/ui/events.ts]: Yea, bad search and replace... :-(

**Pull request merged**